kore/src/Kore/Step/Simplification/Data.hs

```diff
@@ -93,23 +93,30 @@ import           Kore.Internal.Pattern
                  ( Pattern, Predicate )
 import qualified Kore.Internal.Predicate as Predicate
 import           Kore.Internal.TermLike
-                 ( Symbol, TermLike, TermLikeF (..) )
+                 ( SubstitutionVariable, Symbol, TermLike, TermLikeF (..) )
 import           Kore.Logger
 import           Kore.Profiler.Data
                  ( MonadProfiler (profileDuration) )
 import           Kore.Step.Axiom.Identifier
                  ( AxiomIdentifier )
 import           Kore.Syntax.Application
-import           Kore.Syntax.Variable
-                 ( SortedVariable )
-import           Kore.Unparser
 import           Kore.Variables.Fresh
 import           ListT
                  ( ListT (..), mapListT )
 import qualified ListT
 import           SMT
                  ( MonadSMT (..), SMT, SmtT (..) )
 
+{- | 'SimplifierVariable' constrains variables that are used in the simplifier.
+
+'SimplifierVariable' requires only that the variable be a 'SubstitutionVariable'
+(and in turn, an 'InternalVariable'), but simplifier functions should use this
```

Comment:
It has an extra constraint besides `InternalVariable`, i.e. `FreshVariable`, so, at the very least, "either" is a bit confusing below.